build_tools/azure/install.sh

```diff
@@ -56,6 +66,11 @@ pre_python_environment_install() {
         sudo apt-get update
         sudo apt-get install intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic
         source /opt/intel/oneapi/setvars.sh
+    elif [[ "$DISTRIB" == "pip-nogil" ]]; then
+        echo "deb-src http://archive.ubuntu.com/ubuntu/ focal main" | sudo tee -a /etc/apt/sources.list
+        sudo apt-get -yq update
+        sudo apt-get install -yq ccache
+        sudo apt-get build-dep -yq python3 python3-dev
```

Comment:
Shouldn't this go before the previous check, line 61?

Reply:
moved.